saleor/discount/migrations/tasks/saleor3_16.py

```diff
@@ -317,8 +357,14 @@ def migrate_order_line_discounts(sale_ids, rule_by_channel_and_sale):
                         line=order_line,
                     )
                 )
-
+        last_created_at = order_line.created_at
+        already_processed_ids = list(
+            order_lines.filter(created_at=last_created_at).values_list("id", flat=True)
+        )
         OrderLineDiscount.objects.bulk_create(order_line_discounts)
+        migrate_order_line_discounts(
```

Comment:
And same as before possible recursion limit.

Reply:
I changed it to celery task